torch/testing/_internal/common_utils.py

```diff
@@ -780,7 +780,7 @@ def _check_module_exists(name: str) -> bool:
 # TODO: Remove PYTORCH_MIOPEN_SUGGEST_NHWC once ROCm officially supports NHWC in MIOpen
 # See #64427
 TEST_WITH_MIOPEN_SUGGEST_NHWC = os.getenv('PYTORCH_MIOPEN_SUGGEST_NHWC', '0') == '1'
-
+TEST_WITH_MPS = os.getenv('PYTORCH_TEST_WITH_MPS', '0') == '1'
```

Comment:
This I don't understand, we should determine if we test by MPS simply by MPS's availability

Reply:
Do you have a sense for how MPS will handle the tests, yet, @kulinseth? Like is the reason we want this flag for now because so many tests are failing when run on MPS? Are you thinking of maybe gradually opting tests into using MPS or opting out all tests that fail on it?

Reply:
When machines have both the mps and cpu device type, won't a test class be generated for each device type, though?

Skipping is easy with our current architecture, but if an incredible number of skips are required then it may be preferable to opt-in tests to run on MPS by, for example, adding an allow list of tests that can be instantiated for the mps device type.

Reply:
> When machines have both the mps and cpu device type, won't a test class be generated for each device type, though?
@mruberry , this was an mis-understanding on our part. Thanks for clarification and @albanD also helped me see the right place to enable it. 

> 
> Skipping is easy with our current architecture, but if an incredible number of skips are required then it may be preferable to opt-in tests to run on MPS by, for example, adding an allow list of tests that can be instantiated for the mps device type.

Right, we have test_mps.py file we have added (coming up in next PR), which has collection of tests written by us and sourced from different places in PyTorch to smoke-test our functionality. We were thinking we can add skips at the file level to begin with and just test_mps to keep the CI green. And as and when we enable more tests, we can add individual test skips rather than whole file skip.

Reply:
Removed this with a check from torch.backends.mps.is_available()